### ISSUE TITLE
Upgrade CI actions

### ISSUE THREAD

**Issue by turt2live** [matrix-org/matrix-spec#2039]
![image](https://github.com/user-attachments/assets/8e91e07e-ea35-43b4-b3a5-07da1d5bff63)

**Comment by richvdh** [comment]
I don't think this needs any further action. We are now using `ubuntu-24.04` and the world has yet to end.

(The rollout was originally scheduled for October 2024, but github [reverted that](https://github.com/actions/runner-images/issues/10636#issuecomment-2417303444) and it happened in January 2025 instead)
